Data/DataRepository.cs

```diff
@@ -174,23 +196,203 @@ public Comment GetComment(int commentId)
+        {
+            ChangeCourseStatus("published", course_id);
+        }
+
         public Comment PostComment(Comment comment)
         {
             using (var connection = new MySqlConnection(_connectionString))
             {
                 connection.Open();
                 connection.Execute(
-                    @"INSERT INTO Comment( user_id, component_id, text) VALUES (@UserId, @ComponentId, @Text);", 
-                    new {comment.UserId, comment.ComponentId, comment.Text}
+                    @"INSERT INTO Comment( user_id, component_id, `text`, created_at) 
+                      VALUES (@UserId, @ComponentId, @Text, @CreatedAt);",
+                    new { comment.UserId, comment.ComponentId, comment.Text, comment.CreatedAt }
                 );
                 return connection.QueryFirst<Comment>(
-                    @"SELECT id, user_id AS userId, component_id AS componentId, text FROM Comment ORDER BY id DESC LIMIT 1;"
+                    @"SELECT id, user_id AS userId, component_id AS componentId, `text`, created_at AS createdAt 
+                      FROM Comment ORDER BY id DESC LIMIT 1;"
+                );
+            }
+        }
+
+        public void InsertToCategoryCourse(int courseId, int categoryId)
+        {
+            using (var connection = new MySqlConnection(_connectionString))
+            {
+                connection.Open();
+                connection.Execute(
+                    @"INSERT INTO category_course(course_id, category_id) 
+                      VALUES (@CourseId, @CategoryId);",
+                    new { CourseId =courseId, CategotyId = categoryId }
+                );
+            }
+        }
+
+        public Course CreateCourse(Course course)
+        {
+            string status = "dev";
+            using (var connection = new MySqlConnection(_connectionString))
+            {
+                connection.Open();
+                var courseId = Convert.ToInt32(connection.ExecuteScalar(
+                    @"INSERT INTO Course(creator_id, `title`, description, `price`, acceptance_percantage, requirements,
+                        `status`, published_at) 
+                        VALUES(@CreatorId, @Title, @Description, @Price, @AcceptancePercantage, @Requirements, 
+                        @Status, @PublishedAt);
+                        SELECT LAST_INSERT_ID();",
+                    new
+                    {
+                        CreatorId = course.CreatorId,
+                        Title = course.Title,
+                        Description = course.Description,
+                        Price = course.Price,
+                        AcceptancePercantage = course.AcceptancePercantage,
+                        Requirements = course.Requirements,
+                        Status = status,
+                        PublishedAt = DateTime.Now
+                    }
+                ));
+                course.Id = courseId;
```

Comment:
Now I see what you meant when you told me that previous recursive course creating approach was better. My bad.
It's up to you whether leave this or roll back to previous approach. 

But I'd rather leave `SELECT LAST_INSERT_ID();` so as not to make extra DB query for created entity.

Reply:
I haven't changed anything in this moment. Let it stay as it is.